app/scripts/contentscript.js

```diff
@@ -68,122 +71,225 @@ function injectScript(content) {
-      },
-      window.location.origin,
-    );
-  });
+        window.location.origin,
+      );
+    },
+  );
 
   // forward communication across inpage-background for these channels only
-  forwardTrafficBetweenMuxes(PHISHING_SAFELIST, pageMux, extensionMux);
+  forwardTrafficBetweenMuxes(
+    PHISHING_SAFELIST,
+    pagePhishingMux,
+    extensionPhishingMux,
+  );
 }
 
 /**
- * Sets up two-way communication streams between the
- * browser extension and local per-page browser context.
- *
+ * Creates and connects channel multiplexes to handle the channels individually
  */
-async function setupStreams() {
-  // the transport-specific streams for communication between inpage and background
-  const pageStream = new WindowPostMessageStream({
+const setupPageStreams = () => {
+  console.log('setupPageStreams called');
+
+  pageStream = new WindowPostMessageStream({
     name: CONTENT_SCRIPT,
     target: INPAGE,
   });
-  const extensionPort = browser.runtime.connect({ name: CONTENT_SCRIPT });
-  const extensionStream = new PortStream(extensionPort);
 
-  // create and connect channel muxers
-  // so we can handle the channels individually
-  const pageMux = new ObjectMultiplex();
+  pageMux = new ObjectMultiplex();
   pageMux.setMaxListeners(25);
-  const extensionMux = new ObjectMultiplex();
+
+  extensionMux = new ObjectMultiplex();
   extensionMux.setMaxListeners(25);
   extensionMux.ignoreStream(LEGACY_PUBLIC_CONFIG); // TODO:LegacyProvider: Delete
 
   pump(pageMux, pageStream, pageMux, (err) =>
     logStreamDisconnectWarning('MetaMask Inpage Multiplex', err),
   );
+
   pump(extensionMux, extensionStream, extensionMux, (err) => {
     logStreamDisconnectWarning('MetaMask Background Multiplex', err);
     notifyInpageOfStreamFailure();
   });
 
-  // forward communication across inpage-background for these channels only
+  /** forward communication across inpage-background for these channels only */
   forwardTrafficBetweenMuxes(PROVIDER, pageMux, extensionMux);
 
-  // connect "phishing" channel to warning system
-  const phishingStream = extensionMux.createStream('phishing');
+  /** connect "phishing" channel to warning system */
+  phishingStream = extensionMux.createStream('phishing');
   phishingStream.once('data', redirectToPhishingWarning);
+};
 
-  // TODO:LegacyProvider: Delete
-  // handle legacy provider
-  const legacyPageStream = new WindowPostMessageStream({
-    name: LEGACY_CONTENT_SCRIPT,
-    target: LEGACY_INPAGE,
-  });
+/**
+ * Destroying the destination streams of the pump (should) destroy the streams before it
+ *
+ * @todo
+ * - Remove text (should) above after confirming this works
+ * - Answer questions:
+ *   1. is it necessary to manually set allowHalfOpen to false?
+ *      "If false then the stream will automatically end the writable side when the readable side ends"
+ *   2 is it necessary to removeAllListeners if we're calling destroy? - I believe calling destroy
+ *     removes all listeners.
```

Comment:
note: I do not think calling `destroy()` removes its listeners as I ran into the stream disconnect error when commenting out `removeAllListeners();`. Keeping `removeAllListeners();` in does not produce the following console error.

<img width="649" alt="Screenshot 2022-08-19 at 3 25 26 AM" src="https://user-images.githubusercontent.com/20778143/185478702-821b8d89-d6e7-4a02-9661-cc15778ee826.png">